torch/utils/_config_module.py

```diff
@@ -13,6 +13,27 @@
 from typing_extensions import deprecated
 from unittest import mock
 
+from torch._utils_internal import justknobs_check
+
+
+@dataclass
+class Config:
+    """Represents a config with richer behaviour than just a default value.
+
+    i.e.
+    foo = Config(justknob="//foo:bar", default=False)
+    install_config_module(...)
+
+    This configs must be installed with install_config_module to be used
+
+    Arguments:
+        justknob: the name of the feature / JK. In OSS this is unused.
+        default: is the value to default this knob to in OSS.
```

Comment:
You need to have more detail about the precedence order, as that is public API

Reply:
Done.